### PR TITLE
Throw exception if no method is defined on a symfony route

### PR DESCRIPTION
**Pull request by apogadaev**
Fixes protung/open-api-generator#2

### CODE REVIEW

File: src/Processor/Path/Symfony/PathProcessor.php [hunk 1]
```diff
@@ -53,6 +54,7 @@ public function process(Path $path): array
      */
     private function processRoute(SymfonyRoute $route, SymfonyRoutePath $path): array
     {
+        Assert::isNonEmptyList($route->getMethods());
```
Comment (dragosprotung) [inline src/Processor/Path/Symfony/PathProcessor.php:57]: We should throw `InvalidArgumentException` like in `process()` with a custom message.

**Review by dragosprotung — CHANGES_REQUESTED** [review]

File: src/Processor/Path/Symfony/PathProcessor.php [hunk 2]
```diff
@@ -53,6 +53,12 @@ public function process(Path $path): array
      */
     private function processRoute(SymfonyRoute $route, SymfonyRoutePath $path): array
     {
+        if ($route->getMethods() === []) {
+            throw new InvalidArgumentException(
+                Psl\Str\format('The defined methods for the "%s" route do not exist in the API doc configuration.', $path->routeName()),
```
Comment (dragosprotung) [inline src/Processor/Path/Symfony/PathProcessor.php:58]: Message: `No HTTP methods defined for route "%s".`

File: tests/Processor/Symfony/PathProcessorTest.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+use Protung\OpenApiGenerator\Model\ModelRegistry;
+use Protung\OpenApiGenerator\Model\Path\Input\HeaderInput;
+use Protung\OpenApiGenerator\Model\Response;
+use Protung\OpenApiGenerator\Processor\Path\Symfony\PathProcessor;
+use Protung\OpenApiGenerator\Processor\Path\Symfony\SymfonyRoutePath;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollection;
+
+final class PathProcessorTest extends TestCase
```
Comment (dragosprotung) [inline tests/Processor/Symfony/PathProcessorTest.php:24]: Now that we have a unit test for this class, please add tests for the happy path as well

**Review by dragosprotung — CHANGES_REQUESTED** [review]

File: tests/Processor/Symfony/PathProcessorTest.php (hunk 3 above)
Comment (apogadaev) [reply tests/Processor/Symfony/PathProcessorTest.php:24]: @dragosprotung Is it enough to have these asserts in new test?
```php
        $actual = $pathProcessor->process($path);

        self::assertCount(1, $actual);
        self::assertContainsOnly(PathOperation::class, $actual);
```

Comment (dragosprotung) [reply tests/Processor/Symfony/PathProcessorTest.php:24]: We should assert the path operations

File: tests/Processor/Symfony/PathProcessorTest.php [hunk 4]
```diff
@@ -0,0 +1,168 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Protung\OpenApiGenerator\Tests\Processor\Symfony;
```
Comment (dragosprotung) [inline tests/Processor/Symfony/PathProcessorTest.php:5]: Invalid namespace

File: tests/Processor/Symfony/PathProcessorTest.php [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+                $this->createMock(ExampleDescriber::class),
+            ),
+        );
+
+        $path = new SymfonyRoutePath(
+            'test_api.application.ping',
+            'API',
+            'Get information about the API.',
+            null,
+            [HeaderInput::withName('test')],
```
Comment (dragosprotung) [inline tests/Processor/Symfony/PathProcessorTest.php:52]: No need for actual values for exception test

File: tests/Processor/Symfony/PathProcessorTest.php [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+                                ],
+                            ),
+                        ],
+                    ),
+                    'security' => [],
+                ],
+            ),
+        );
+
+        self::assertCount(1, $actual);
```
Comment (dragosprotung) [inline tests/Processor/Symfony/PathProcessorTest.php:165]: Why not full assert of the actual ?

File: tests/Processor/Symfony/PathProcessorTest.php [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+                new ObjectDescriber(
+                    new ModelRegistry(),
+                    $this->createMock(ObjectDescriber\Describer::class),
+                ),
+                new FormFactory($this->createMock(FormFactoryInterface::class)),
+                $this->createMock(ExampleDescriber::class),
+            ),
+        );
+
+        $path = new SymfonyRoutePath(
```
Comment (dragosprotung) [inline tests/Processor/Symfony/PathProcessorTest.php:92]: Construct a maximal path

**Review by dragosprotung — CHANGES_REQUESTED** [review]

**Review by dragosprotung — APPROVED** [review]
Thank you @apogadaev

**Pull request merged**